src/sentry/deletions/defaults/detector.py

```diff
@@ -0,0 +1,31 @@
+from sentry.deletions.base import BaseRelation, ModelDeletionTask, ModelRelation
+from sentry.workflow_engine.models.detector import Detector
+
+
+class DetectorDeletionTask(ModelDeletionTask[Detector]):
+    def get_child_relations(self, instance: Detector) -> list[BaseRelation]:
```

Comment:
should we also delete `DetectorWorkflow` if that relationship exists? as well as related lookup tables (e.g. `AlertRuleDetector`?)

Reply:
`DetectorWorkflow` does get deleted automatically via cascade delete, the test checks for it. I'm not sure that we want to delete the migration tables data here since that will be temporary.

Reply:
Just note that with cascade delete if there are a large number of related rows the delete can fail. That's probably unlikely here, but jfyi